account/src/main/java/com/user/account/controller/AllControlers.java

```diff
@@ -28,38 +49,56 @@ public String home() {
 		// Updating accountBalance after debit
 		
 		
-		@PostMapping("/debit/{accountNumber}/{amount}")
-		public User debit(@PathVariable String accountNumber, @PathVariable String amount) {
+		@PostMapping("/debit/{accountNumber}")
+		public void debit(@PathVariable String accountNumber, @RequestBody String amount) {
+
+			long userAccountNumber = Long.parseLong(accountNumber);
+
+			long debitAmount = Long.parseLong(amount);
 			
-			User user = this.accountServices.getUser(Long.parseLong(accountNumber));
+			User user = accountService.getUser(userAccountNumber);
 			
-			if ( Long.parseLong(amount) <= user.getAccountBalance() ) {
+			if ( debitAmount <= user.getAccountBalance() ) {
```

Comment:
For comparing two Objects in JAVA please use `compareTo`

Ref: https://docs.oracle.com/javase/7/docs/api/java/lang/Long.html#compareTo(java.lang.Long)